Make JavaExec default to the java toolchain (#33264)
gradle/gradle: platforms/documentation/docs/src/docs/userguide/releases/upgrading/upgrading_version_8.adoc

JaCoCo has been updated to https://www.jacoco.org/jacoco/trunk/doc/changes.html[0.8.13].

> ==== The `JavaExec` task now defaults to the toolchain configured on the `java` extension

Reviewer: ```suggestion
==== `JavaExec` now uses the toolchain from the `java` extension by default
```

> Previously, the `JavaExec` task would default to use the Java version of the current Gradle process.

Reviewer: ```suggestion
Previously, the `JavaExec` task used the same Java version as the Gradle process itself.
```

> Now, it will default to the toolchain set in the `java` extension whenever the `java-base` plugin has been applied.

Reviewer: ```suggestion
Starting in Gradle 9.0, when the `java-base` plugin is applied, `JavaExec` will instead default to the Java toolchain configured in the `java` extension.
```

> As always, this default can be overridden in the `JavaExec` task configuration to another toolchain as necessary.

Reviewer: ```suggestion
You can override the toolchain explicitly in the `JavaExec` task configuration if needed.
```
